addons/website_mail/controllers/main.py

```diff
@@ -76,55 +75,50 @@ class WebsiteMail(http.Controller):
     @http.route(['/website_mail/follow'], type='json', auth="public", website=True)
     def website_message_subscribe(self, id=0, object=None, message_is_follower="on", email=False, **post):
         # TDE FIXME: check this method with new followers
-        cr, uid, context = request.cr, request.uid, request.context
-
-        partner_obj = request.registry['res.partner']
-        user_obj = request.registry['res.users']
+        Partner = request.env['res.partner'].sudo()
 
         _id = int(id)
         _message_is_follower = message_is_follower == 'on'
-        _object = request.registry[object]
+        _object = request.env[object]
 
         # search partner_id
         public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_ids = [user_obj.browse(cr, uid, uid, context).partner_id.id]
+        if request.env.uid != public_id:
+            partner_ids = [request.env.user.partner_id.id]
         else:
             # mail_thread method
-            partner_ids = _object._find_partner_from_emails(
-                cr, SUPERUSER_ID, [_id], [email], context=context, check_followers=True)
+            partner_ids = _object.sudo()._find_partner_from_emails([email], check_followers=True)
```

Comment:
you are suppose to call _find_partner_from_emails(...) on a recordset instead of class instance.